src/test/java/org/javarosa/entities/internal/EntityFormParseProcessorTest.java

```diff
@@ -25,6 +26,76 @@
 
 public class EntityFormParseProcessorTest {
 
+    @Test(expected = UnrecognizedEntityVersionException.class)
+    public void whenVersionIsNotRecognized_throwsException() throws IOException {
+        XFormsElement form = XFormsElement.html(
+            asList(
+                new Pair<>("entities", "http://www.opendatakit.org/xforms/entities")
+            ),
+            head(
+                title("Create entity form"),
+                t("model entities:entities-version=\"somethingElse\"",
+                    mainInstance(
+                        t("data id=\"create-entity-form\"",
+                            t("name"),
+                            t("orx:meta",
+                                t("entities:entity dataset=\"people\"",
+                                    t("entities:create")
+                                )
+                            )
+                        )
+                    ),
+                    bind("/data/name").type("string").withAttribute("entities", "saveto", "name")
+                )
+            ),
+            body(
+                input("/data/name")
+            )
+        );
+
+        EntityFormParseProcessor processor = new EntityFormParseProcessor();
+
+        XFormParser parser = new XFormParser(new InputStreamReader(new ByteArrayInputStream(form.asXml().getBytes())));
+        parser.addProcessor(processor);
+        parser.parse(null);
+    }
+
+    @Test
+    public void whenVersionIsNewPatch_doesNotThrowException() throws IOException {
+        String newPatchVersion = EntityFormParseProcessor.SUPPORTED_VERSION + ".12";
+
+        XFormsElement form = XFormsElement.html(
+            asList(
+                new Pair<>("entities", "http://www.opendatakit.org/xforms/entities")
+            ),
+            head(
+                title("Create entity form"),
+                t("model entities:entities-version=\"" + newPatchVersion + "\"",
+                    mainInstance(
+                        t("data id=\"create-entity-form\"",
+                            t("name"),
+                            t("orx:meta",
+                                t("entities:entity dataset=\"people\"",
+                                    t("entities:create")
+                                )
+                            )
+                        )
+                    ),
+                    bind("/data/name").type("string").withAttribute("entities", "saveto", "name")
+                )
+            ),
+            body(
+                input("/data/name")
+            )
+        );
+
+        EntityFormParseProcessor processor = new EntityFormParseProcessor();
+
+        XFormParser parser = new XFormParser(new InputStreamReader(new ByteArrayInputStream(form.asXml().getBytes())));
+        parser.addProcessor(processor);
+        parser.parse(null);
```

Comment:
You don't want to add any kind of assertion here? I think this is fine but I feel like I've seen you throw back tests like this so want to give you a chance to consider it again in case you do care.

Reply:
I guess it came from writing these out in order (in a TDD red-green cycle). The only needed a test where something didn't explode to get the code where I wanted, but you're right that it's a bit confusing to read after the fact. I'll add a "this is working as expected" assertion.